ui5/browser/controller/Browser.controller.js

```diff
@@ -283,34 +284,40 @@ sap.ui.define(['sap/ui/core/mvc/Controller',
          return this.websocket.Send("SAVEFILE:" + fullpath + ":" + sText);
       },
 
-      reallyRunMacro: function () {
-         const oEditor = this.getSelectedCodeEditor();
+      reallyRunMacro: function (myThis) {
+         const oEditor = myThis.getSelectedCodeEditor();
          const oModel = oEditor.getModel();
          const fullpath = oModel.getProperty("/fullpath");
          if (fullpath === undefined)
-            return this.onSaveAs();
-         return this.websocket.Send("RUNMACRO:" + fullpath);
+            return myThis.onSaveAs();
+         return myThis.websocket.Send("RUNMACRO:" + fullpath);
       },
 
       /** @brief Handle the "Run" button press event */
       onRunMacro: function () {
+         this.saveCheck(this.reallyRunMacro);
```

Comment:
Here one should use `bind` for function like:
```
this.saveCheck(this.reallyRunMacro.bind(this));
```
In such case `myThis` logic is not necessary.

Reply:
Ok i will try

Reply:
This is done